mono/metadata/cominterop.c

```diff
@@ -2359,12 +2359,24 @@ cominterop_get_managed_wrapper_adjusted (MonoMethod *method)
 
 	if (!MONO_TYPE_IS_VOID (sig->ret)) {
 		if (!preserve_sig) {
-			MonoClass *rclass = mono_class_from_mono_type_internal (sig->ret);
+			int pos_null;
+			MonoClass *rclass;
+
+			mono_mb_emit_stloc (mb, retval);
+			mono_mb_emit_ldarg (mb, sig_native->param_count-1);
+			pos_null = mono_mb_emit_branch (mb, CEE_BRFALSE);
+
+			mono_mb_emit_ldarg (mb, sig_native->param_count-1);
+			mono_mb_emit_ldloc (mb, retval);
+
+			rclass = mono_class_from_mono_type_internal (sig->ret);
```

Comment:
Is there reason this separate from local variable?

Reply:
Do you mean the C89-style declaration? No particular reason, I guess; it seems Mono usually uses C89-style declarations, except when it doesn't, and I'm personally more used to C89.

Reply:
Yes. Mono is very C99. Uninitialized locals are bug farms imho.
Though I agree C99 is hardly worth anything over C89.
I'd prefer C++ even C++96.

Reply:
Okay, fixed.